### PR TITLE
Fix some problems with file upload

### PR DESCRIPTION
**Pull request by Atmos4**
- missing entry in .htaccess
- simplify some complexity
- extracted render components
- add disclaimer about supported file formats

### CODE REVIEW

File: app/pages/shared_documents/renderDocument.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    $file_mime = $shared_doc->mime;
+
+    $file_icon = match ($file_mime) {
+        'pdf' => 'fas fa-file-pdf',
+        'png', 'jpg', 'jpeg', 'gif' => 'fas fa-file-image',
+        'doc', 'docx' => 'fas fa-file-word',
+        'xls', 'xlsx' => 'fas fa-file-excel',
+        'ppt', 'pptx' => 'fas fa-file-powerpoint',
+        default => 'fas fa-file',
+    };
```
Comment (Atmos4) [inline app/pages/shared_documents/renderDocument.php:14]: `match` is a very powerful operator. 7 lines of codes compared to 25 with the previous `switch` code

File: app/pages/shared_documents/renderDocument.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                <i class="fas fa-trash"></i>
+            </a>
+        </td>
+    </tr>
+<?php }
+
+function renderTable($title, $docs)
+{
+    if (!$docs)
+        return; ?>
```
Comment (Atmos4) [inline app/pages/shared_documents/renderDocument.php:40]: returning early like this is very powerful. It reduces nesting and makes the code very easy to understand.

short-circuiting functions like this one are also less bug prone.

File: database/models/files.db.php [hunk 3]
```diff
@@ -51,4 +51,10 @@ static function get($file_id): SharedFile|null
         return em()->find(SharedFile::class, $file_id);
     }
 
+    /** @return self[] */
+    static function findBy($permission): array
+    {
+        return em()->getRepository(self::class)->findBy(['permission_level' => $permission]);
+    }
```
Comment (Atmos4) [inline database/models/files.db.php:58]: extracting this logic allows to have type safety! Then you get IDE suggestions from the SharedFile class.

I believe there is some way to get generics to work with Psalm... maybe we should look into it 🤔

**Comment by Atmos4** [comment]
Hope the above comments help you in your code review. Let me know if you want to talk more about certain aspects of the refactoring

**Review by Mathieu414 — APPROVED** [review]
Good job 🫶

**Pull request merged**